org.eclipse.xtext/src/org/eclipse/xtext/linking/lazy/LazyLinkingResource.java

```diff
@@ -287,7 +305,10 @@ protected EObject getEObject(String uriFragment, Triple<EObject, EReference, INo
 			createAndAddDiagnostic(triple, ex);
 			return null;
 		} finally {
-			resolving.remove(triple);
+			if (cyclicLinkingDetectionCounter > cyclicLinkingDectectionCounterLimit) {
+                resolving.remove(triple);
+            }
+            cyclicLinkingDetectionCounter--;
```

Comment:
Indentation looks weird. Mixed tabs / spaces? Maybe reformat that method?

Reply:
fixed